components/layout/construct.rs

```diff
@@ -222,7 +248,7 @@ impl InlineFragmentsAccumulator {
                 pseudo: node.get_pseudo_element_type().strip(),
                 style: node.style(style_context),
                 selected_style: node.selected_style(),
-                flags: InlineFragmentNodeFlags::empty(),
+                flags: FIRST_FRAGMENT_OF_ELEMENT | LAST_FRAGMENT_OF_ELEMENT,
```

Comment:
I think of InlineFragmentsAccumulator's `enclosing_node` as representing the "whole" InlineFragmentNodeInfo for this accumulator, i.e. what it would be if it was all one fragment. It's split in two each time an {ib} split is encountered, then split up again in `to_intermediate_inline_fragments`. I'm not convinced that it's conceptually more reasonable to save instructions for how to mark the `enclosing_nodes` and follow them as late as possible rather than simply handling it this way as we go.

Practically speaking, Servo will one day have to handle the positioning of backgrounds across multiple line boxes. Information about this will likely be stored in InlineFragmentNodeInfo. Using a conceptual model that allows that information to be reused will make it easier.

Comments would definitely help, though. If we stick with my model, I'll add some explaining it.